apps/api/src/auth/services/api-key/api-key-auth.service.spec.ts

```diff
@@ -0,0 +1,140 @@
+import { container } from "tsyringe";
+
+import { ApiKeyOutput, ApiKeyRepository } from "@src/auth/repositories/api-key/api-key.repository";
+import { ApiKeyAuthService } from "./api-key-auth.service";
+import { ApiKeyGeneratorService } from "./api-key-generator.service";
+
+jest.mock("@src/auth/repositories/api-key/api-key.repository");
+
+describe("ApiKeyAuthService", () => {
+  let service: ApiKeyAuthService;
+  let apiKeyGenerator: ApiKeyGeneratorService;
+  let apiKeyRepository: jest.Mocked<ApiKeyRepository>;
+  let originalEnv: string | undefined;
+
+  beforeEach(() => {
+    originalEnv = process.env.NODE_ENV;
+    apiKeyGenerator = container.resolve(ApiKeyGeneratorService);
+
+    apiKeyRepository = {
```

Comment:
[this](https://github.com/akash-network/console/blob/main/apps/api/test/services/stub.ts) might be useful
```suggestion
    apiKeyRepository = stub<ApiKeyRepository>({ findOneBy: jest.fn() })
```